contracts/JBController.sol

```diff
@@ -372,46 +360,46 @@ contract JBController3_1 is
         // Keep a reference to the reserved rate to use
         uint256 _reservedRate;
 
-        // Scoped section prevents stack too deep. `_fundingCycle` only used within scope.
+        // Scoped section prevents stack too deep. `_ruleset` only used within scope.
         {
-            // Get a reference to the project's current funding cycle.
-            JBFundingCycle memory _fundingCycle = fundingCycleStore.currentOf(_projectId);
+            // Get a reference to the project's current ruleset.
+            JBRuleset memory _ruleset = rulesets.currentOf(_projectId);
 
-            // Minting limited to: project owner, authorized callers, project terminal and current funding cycle data source
+            // Minting limited to: project owner, operators with the `MINT_TOKENS` permission from the owner, the project's terminals, or the project's current ruleset data hook
             _requirePermissionAllowingOverride(
                 projects.ownerOf(_projectId),
                 _projectId,
-                JBOperations.MINT_TOKENS,
-                directory.isTerminalOf(_projectId, IJBPaymentTerminal(_msgSender()))
-                    || _msgSender() == address(_fundingCycle.dataSource())
+                JBPermissionIds.MINT_TOKENS,
+                directory.isTerminalOf(_projectId, IJBTerminal(_msgSender()))
+                    || _msgSender() == address(_ruleset.dataHook())
             );
 
-            // If the message sender is not a terminal or a datasource, the current funding cycle must allow minting.
+            // If the message sender is not a terminal or a data hook, the current ruleset must allow minting.
             if (
-                !_fundingCycle.mintingAllowed()
-                    && !directory.isTerminalOf(_projectId, IJBPaymentTerminal(_msgSender()))
-                    && _msgSender() != address(_fundingCycle.dataSource())
-            ) revert MINT_NOT_ALLOWED_AND_NOT_TERMINAL_DELEGATE();
+                !_ruleset.allowDiscretionaryMinting()
```

Comment:
@ who's discretion? Is it worth being more specific here i.e. "allowAdminMinting()"?

Reply:
nice

Reply:
Going with allowOwnerMinting. Nice